hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -827,8 +827,9 @@ protected void putAllInternal(final Map<? extends Object, ? extends Object> entr
         int partitionCount = partitionService.getPartitionCount();
         boolean tooManyEntries = entries.size() > (partitionCount * factor);
         try {
+            List<Future<?>> futures = new LinkedList<Future<?>>();
+
```

Comment:
Gladly. I share your opinions, though with this PR I was looking to preserve code flow. Given that you've now created the new PR that embodies your suggestions, I'll close this one.